executor/runtime/docker/docker.go

```diff
@@ -947,35 +943,14 @@ func (r *DockerRuntime) Prepare(parentCtx context.Context) error { // nolint: go
 		return nil
 	})
 
-	if shouldStartSpectatord(&r.cfg, r.c) {
-		group.Go(r.createVolumeContainerFunc(sidecarConfigs[runtimeTypes.SidecarServiceSpectatord], &spectatordContainerName))
-	}
-
-	if shouldStartAtlasd(&r.cfg, r.c) {
-		group.Go(r.createVolumeContainerFunc(sidecarConfigs[runtimeTypes.SidecarServiceAtlasd], &atlasdContainerName))
-	}
-
-	if shouldStartMetatronSync(&r.cfg, r.c) {
-		group.Go(r.createVolumeContainerFunc(sidecarConfigs[runtimeTypes.SidecarServiceMetatron], &metatronContainerName))
-	}
-
-	if shouldStartSSHD(&r.cfg, r.c) {
-		group.Go(r.createVolumeContainerFunc(sidecarConfigs[runtimeTypes.SidecarServiceSshd], &sshdContainerName))
-	}
-
-	if shouldStartLogViewer(&r.cfg, r.c) {
-		group.Go(r.createVolumeContainerFunc(sidecarConfigs[runtimeTypes.SidecarServiceLogViewer], &logViewerContainerName))
-	}
-
-	if shouldStartServiceMesh(&r.cfg, r.c) {
-		group.Go(r.createVolumeContainerFunc(sidecarConfigs[runtimeTypes.SidecarServiceServiceMesh], &serviceMeshContainerName))
-	}
-
-	if shouldStartAbmetrix(&r.cfg, r.c) {
-		group.Go(r.createVolumeContainerFunc(sidecarConfigs[runtimeTypes.SidecarServiceAbMetrix], &abmetrixContainerName))
+	for sidecarName, sidecarConfig := range sidecarConfigs {
+		if sidecarConfig.Volumes != nil && sidecarConfig.EnabledCheck(&r.cfg, r.c) {
+			sidecarConfig.ContainerName = sidecarName
+			group.Go(r.createVolumeContainerFunc(sidecarName, sidecarConfig))
```

Comment:
Is this valid, or does it create a pointer leak?

Reply:
I was very confused about this pointer stuff.
I don't really know the answer. Somehow we go from a "base" container name to an expanded one that docker will like.
The code did seem work find in this PR, I don't know if there is a leak or not. If there is it will only be a few here for the lifetime of the task?

Reply:
Can you clarify though (it may be outdated now with the changes in the PR) how a leak would happen?

Each `sidecarConfig` has a `ContainerName`, should be self-contained?